include/podio/GenericParameters.h

```diff
@@ -162,6 +160,19 @@ class GenericParameters {
     }
   }
 
+  DEPR_NON_TEMPLATE const auto& getStringMap() const {
+    return getMap<std::string>();
+  }
+
+  DEPR_NON_TEMPLATE const auto& getFloatMap() const {
+    return getMap<float>();
+  }
+
+  DEPR_NON_TEMPLATE const auto& getIntMap() const {
+    return getMap<int>();
+  }
+
+private:
   /// Get a reference to the internal map for a given type (necessary for SIO)
```

Comment:
Why is the non const version needed for SIO?

Reply:
It currently populates the maps in place.